staging/src/k8s.io/legacy-cloud-providers/azure/clients/securitygroupclient/azure_securitygroupclient_test.go

```diff
@@ -26,18 +26,131 @@ import (
 	"io/ioutil"
 	"net/http"
 	"testing"
+	"time"
 
 	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-06-01/network"
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/golang/mock/gomock"
 	"github.com/stretchr/testify/assert"
 
+	"k8s.io/client-go/util/flowcontrol"
 	azclients "k8s.io/legacy-cloud-providers/azure/clients"
 	"k8s.io/legacy-cloud-providers/azure/clients/armclient"
 	"k8s.io/legacy-cloud-providers/azure/clients/armclient/mockarmclient"
+	"k8s.io/legacy-cloud-providers/azure/retry"
 )
 
+// 2065-01-24 05:20:00 +0000 UTC
+func getFutureTime() time.Time{
+	return time.Unix(3000000000, 0)
+}
+
+func TestNew(t *testing.T) {
+	config := &azclients.ClientConfig{
+		SubscriptionID:		 "sub",
+		ResourceManagerEndpoint: "endpoint",
+		Location:		 "eastus",
+		RateLimitConfig: &azclients.RateLimitConfig{
+			CloudProviderRateLimit:		   true,
+			CloudProviderRateLimitQPS:	   0.5,
+			CloudProviderRateLimitBucket:	   1,
+			CloudProviderRateLimitQPSWrite:    0.5,
+			CloudProviderRateLimitBucketWrite: 1,
+		},
+		Backoff: &retry.Backoff{Steps: 1},
+	}
+
+	nsgClient := New(config)
+	assert.Equal(t, "sub", nsgClient.subscriptionID)
+	assert.NotEmpty(t, nsgClient.rateLimiterReader)
+	assert.NotEmpty(t, nsgClient.rateLimiterWriter)
+}
+
+func TestGet(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	resourceID := "/subscriptions/subscriptionID/resourceGroups/rg/providers/Microsoft.Network/networkSecurityGroups/nsg1"
+	response := &http.Response{
+		StatusCode: http.StatusOK,
+		Body:	    ioutil.NopCloser(bytes.NewReader([]byte("{}"))),
+	}
+
+	armClient := mockarmclient.NewMockInterface(ctrl)
+	armClient.EXPECT().GetResource(gomock.Any(), resourceID, "").Return(response, nil).Times(1)
+	armClient.EXPECT().CloseResponse(gomock.Any(), gomock.Any()).Times(1)
+
+	nsgClient := getTestSecurityGroupClient(armClient)
+	result, rerr := nsgClient.Get(context.TODO(), "rg", "nsg1", "")
+	assert.NotNil(t, result)
```

Comment:
could you check the expected values for the result?

Reply:
sure, will add it.

Reply:
this case seems get an non-exist security group, since I didn't create one. It seems a bug? as you see no err here, and I check return data is nil as following, any idea?
```
                                 ID: (*string)(<nil>),
                                 Name: (*string)(<nil>),
                                  Type: (*string)(<nil>),
                                 Location: (*string)(<nil>),
                                  Tags: (map[string]*string) <nil>

```

Reply:
Get returns type `network.SecurityGroup`, so expected value should be `network.SecurityGroup{}`

Reply:
got you, thanks.